### PR TITLE
Fix tests writing to current work dir

### PR DESCRIPTION
**Pull request by mshabunin**
Several tests were writing files in the current work directory and did not clean up after test. Moved all temporary files to the `/tmp` dir and added a cleanup code.

### CODE REVIEW

File: modules/core/test/test_io.cpp [hunk 1]
```diff
@@ -829,18 +839,6 @@ TEST(Core_InputOutput, filestorage_base64_valid_call)
...
-        "core_io_base64_other_test.json?base64",
-        0
-    };
-    char const * real_name[] = {
-        "core_io_base64_other_test.yml",
-        "core_io_base64_other_test.xml",
-        "core_io_base64_other_test.json",
         "core_io_base64_other_test.yml",
         "core_io_base64_other_test.xml",
         "core_io_base64_other_test.json",
```
Comment (asmorkalov) [inline modules/core/test/test_io.cpp:844]: Why do you remove it?

Comment (mshabunin) [reply modules/core/test/test_io.cpp:844]: In this test we iterate over 3 formats (yml, xml,json) and 2 modes (default, base64). These two arrays contained 6 items each for actual file name and accessed file name (with base64 suffix). I've replaced these arrays with calculated values: `name = filename[n/2]`, `suffix = n%2 ? '' : 'base64'`.

**Comment by Akkshat0420** [comment]
i tried every thing even that change file location D drive to  C Drive .
C:\Users\HARSHIT\Face_recognisation\Scripts\python.exe "D:/SEMESTER 5/Face_recognisation/student.py" 
[ERROR:0@21.929] global obsensor_uvc_stream_channel.cpp:156 cv::obsensor::getStreamChannelGroup Camera index out of range
I found this error.
Anyone have solution please give answere

**Review by asmorkalov — APPROVED** [review]
:+1:

**Pull request merged**